File: Flutter Frontend/smart_class/lib/models/attendance_model.dart
class AttendanceModel{
  String deviceId;
  String student;
  String myClass;

  AttendanceModel({
    required this.student,
    required this.myClass,
    required this.deviceId,
});

  Map<String, dynamic> toJson()
  {
    return {
      "deviceId" : deviceId,
      "student" : student,
      "myClass" :  myClass
    };
  }

  factory AttendanceModel.fromJson(Map<String, dynamic> json)
  {
    return AttendanceModel(student: json['student'],  myClass: json['myClass'], deviceId: json['deviceId']);
  }
}

File: Backend/models/attendance_model.js
const mongoose = require("mongoose");


const attendanceModel = mongoose.Schema({
    deviceId : String,
    student : {
        type : mongoose.Schema.Types.ObjectId,
        ref : "students",
    },
    myClass : {
        type : mongoose.Schema.Types.ObjectId,
        ref : "Classes"
    }
});

module.exports = mongoose.model("attendance", attendanceModel);